proxy/ProxyClientSession.cc

```diff
@@ -142,7 +141,9 @@ ProxyClientSession::do_api_callout(TSHttpHookID id)
   this->api_current = NULL;
 
   if (this->hooks_on && this->has_hooks()) {
-    SET_HANDLER(&ProxyClientSession::state_api_callout);
+    if (!this->handler) {
```

Comment:
Hmm.  It looks like Http1ClientSession::start does not set the handler.  I could add the handler check there and remove the null check here.